lib/link_checker.rb

```diff
@@ -0,0 +1,86 @@
+class LinkChecker
+  class << self
+    def extract_urls(data)
+      new(data).extract_urls
+    end
+
+    def link_checker_api_bearer_token
+      ENV["LINK_CHECKER_API_BEARER_TOKEN"]
+    end
+
+    def for_files_at(path)
+      path = File.join(path, "**/*.erb") unless path.to_s.include?("*")
+      data = Dir.glob(path).map { |filename| File.read(filename) }
+      new(data)
+    end
+  end
+
+  attr_reader :data
+
+  def initialize(data)
+    @data = data
+  end
+
+  def urls
+    # Gather urls in a Set to allow simple combination and duplicate removal
+    @urls ||= extract_urls.to_set
+  end
+
+  def extract_urls
+    case data
+    when String
+      uri = URI.extract(data, %w[http https])
+
+      # markdown urls trailing bracket included in uri so removing here
+      uri.map { |u| unmatched_closing_bracket?(u) ? u.gsub(/\)\W?$/, "") : u }
```

Comment:
Might be worth extracting this out into its own method, so it easy to unit test.